Admin Create User + Edit User implimented

Snail51 opened the pull request:
Addresses #5 

Added / implemented user add/edit pages for admins only.

This solution works, and I have tested it thoroughly, but my implementation may have room for improvement. Please let me know if anything should be changed.

Snail51 committed:
users are creatable but isadmin is not being set

idk why the isadmin parameter is not being set, ill look into it more tomorrow

NOTE:  UserController.php> create is currently arbirarily inverted for testing, and must be reverted before prod

Snail51 committed:
crude user adding implimented

users can now be added, and admins can be declared

will impliment all other feilds next

Snail51 committed:
robust user add menu

now provides fields for active, password confirm,  dept id, and sector id

Snail51 committed:
add "required value" asterisks to user create page

new component `required-asterisk` which is a little red asterisks that onhover says "Required Value". implimentation works but is probably not ideal

Snail51 committed:
hide "add user" button if not signed in as admin

even if you were to click the button as a non-admin, you wouldn't be able to do anything, this just  tidys the UI

Snail51 committed:
fully impliment user editing for admins

edit form takes in necessary inputs and can call UserController::update proper;y

Snail51 committed:
revert unnecessary changes

drraccoony committed:
Merge branch 'main' into admin_create-user

drraccoony commented on app/Http/Controllers/UserController.php:
```diff
@@ -27,25 +32,55 @@ public function index(Request $request)
         return view('users.index', [
-            'users'     => $users
+            'users'     => $users,
+            'viewer'    => $request->user()
```
This is the right idea but this should be removed as the blade can use the `Auth` facade to do the same thing.

drraccoony commented on resources/views/components/required-asterisk.blade.php:
```diff
@@ -0,0 +1,41 @@
+    .hover-element:hover ~ .tooltiptext {
+        visibility: visible;
+        opacity: 1;
+    }
```
We have tailwind, We should use tailwind classes and avoid using any styling being declared like this. The only time we should use styles like this is if there are no tailwind classes.

Snail51 replied:
i reverted this to just be `<b style="color:red">*</b>` for now.

drraccoony commented on resources/views/users/create.blade.php:
```diff
@@ -1,11 +1,16 @@
-        {{ __('Create New User') }}
+        Create a new User
+        <h2>
+            You are currently signed in as <b>Administrator {{$user->name}}</b>
```
You can get the name of the current logged in user with `{{Auth::user()->name}}`.
https://laravel.com/docs/11.x/authorization#via-blade-templates

drraccoony commented on resources/views/users/partials/create-user-form.blade.php:
```diff
@@ -0,0 +1,109 @@
+                <option value="1" {{ old('active', $user->active) == 1 ? 'selected' : '' }}>Active</option>
+                <option value="0" {{ old('active', $user->active) == 0 ? 'selected' : '' }}>Inactive</option>
+            </x-select-input>
+            <x-form-validation for="active" />
+            <x-input-error class="mt-2" :messages="$errors->get('active')" />
```
`<x-form-validation for="email" />` was already created as a component to handle validation errors.

Snail51 replied:
my understanding of this is that we should use <x-form-validation> in all cases instead of <x-input-error>?

drraccoony replied:
Did the input error component already exist?

Snail51 replied:
I don't understand. I borrowed the <x-input-error> from elsewhere in the code, <x-form-validation> is present elsewhere. There are plenty of places where both are used, even beyond what I have touched.

drraccoony reviewed (changes requested):
A great start for just getting into Laravel! Though there are a view things that should be looked at before we merge into `main`.

Snail51 committed:
address all points in #12

- migration to Auth::user instead of passing in $viewer
- using Auth::user, remaining admin-only elements are hidden when signed-in user is not an admin
- made the required asterisk just a red <b>*</b> (ill come back to this later)
- updated creation form to poll departments and sectors like edit form
- migrated all <x-input-error>s to <x-form-validation>s (at least in what i've done)

drraccoony commented on app/Http/Controllers/UserController.php:
```diff
@@ -65,36 +102,42 @@ public function show(string $id)
-    public function update(Request $request, string $id)
+    public function update(Request $request, string $id) : RedirectResponse
     {
+        //dd($request);
+
```
For cleanliness, this should be removed.

drraccoony commented on resources/views/components/checkbox-input.blade.php:
```diff
@@ -0,0 +1,20 @@
+This works because when an HTML form has multiple inputs with the same name, only the last one is submitted to the server.
+-->
+<input type="hidden" name="{{ $name }}" value="0">
+@if ($checked)
+    <input type="checkbox" name="{{ $name }}" value="1" checked class="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-700 dark:border-gray-600">
+@else
+    <input type="checkbox" name="{{ $name }}" value="1" unchecked class="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-700 dark:border-gray-600">
+@endif
```
The proper way of doing this would be:
```blade
<input type="hidden" name="{{ $name }}" value="0">
<input type="checkbox" name="{{ $name }}" value="1" {{ $checked ? 'checked' : '' }} class="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-700 dark:border-gray-600">
```
Or play with the @checked directive. https://laravel.com/docs/11.x/blade#additional-attributes

drraccoony reviewed (changes requested):
Looks great, just these two and Its ready to merge.

Snail51 committed:
address point in #12

addresses https://github.com/minnesota-furs/mnfursvolunteers/pull/12#pullrequestreview-2364952142

Snail51 commented:
> Looks great, just these two and Its ready to merge.

These are now addressed in https://github.com/minnesota-furs/mnfursvolunteers/pull/12/commits/8634b8acdac4a48d80c05bbe4259286d70039156
